src/authV2/azext_authV2/custom.py

```diff
@@ -396,17 +451,53 @@ def get_aad_settings(cmd, resource_group_name, name, slot=None):
 
 def update_aad_settings(cmd, resource_group_name, name, slot=None,  # pylint: disable=unused-argument
                         client_id=None, client_secret_setting_name=None,  # pylint: disable=unused-argument
-                        issuer=None, allowed_token_audiences=None, client_secret=None, yes=False):    # pylint: disable=unused-argument
+                        issuer=None, allowed_token_audiences=None, client_secret=None,  # pylint: disable=unused-argument
+                        yes=False, tenant_id=None, cloud=None):    # pylint: disable=unused-argument
     if client_secret is not None and client_secret_setting_name is not None:
         raise CLIError('Usage Error: --client-secret and --client-secret-setting-name cannot both be '
                        'configured to non empty strings')
 
+    if issuer is not None and (tenant_id is not None or cloud is not None):
+        raise CLIError('Usage Error: --issuer and the combination of --tenant-id and/or --cloud cannot '
+                       'be configured to non empty strings at the same time.')
+
+    is_new_aad_app = False
+    existing_auth = get_auth_settings_v2(cmd, resource_group_name, name, slot)["properties"]
+    registration = {}
+    validation = {}
+    if "identityProviders" not in existing_auth.keys():
+        existing_auth["identityProviders"] = {}
+    if "azureActiveDirectory" not in existing_auth["identityProviders"].keys():
+        existing_auth["identityProviders"]["azureActiveDirectory"] = {}
+        is_new_aad_app = True
+
+    if is_new_aad_app and issuer is None and tenant_id is None:
+        raise CLIError('Usage Error: Either --issuer or --tenant-id must be specified when configuring the '
+                       'Microsoft auth registration.')
+
     if client_secret is not None and not yes:
         msg = 'Configuring --client-secret will add app settings to the web app. Are you sure you want to continue?'
         if not prompt_y_n(msg, default="n"):
             raise CLIError('Usage Error: --client-secret cannot be used without agreeing to add app settings '
                            'to the web app.')
 
+    openid_issuer = issuer
+    if openid_issuer is None:
+        # cmd.cli_ctx.cloud resolves to whichever cloud the customer is currently logged into
+        authority = cmd.cli_ctx.cloud.endpoints.active_directory
+        if cloud is not None:
```

Comment:
Same comment cloud is already set/default to public & one needs to change this to a different cloud type, so whole switch should not be needed.